### PR TITLE
enable lint avoid_classes_with_only_static_members

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter/lib/src/gestures/converter.dart [hunk 1]
```diff
@@ -31,6 +31,7 @@ class _PointerState {
   Offset lastPosition;
 }
 
+// ignore: avoid_classes_with_only_static_members
```
Comment (a14n) [inline packages/flutter/lib/src/gestures/converter.dart]: Without this ignore we'd need to break the API and to find a good top level name for `PointerEventConverter.expand` (`expand` is too general imho).

**Comment by Hixie** [comment]
I don't understand the purpose of this lint. What's wrong with classes with only static members? We use them all over the place for namespacing. In fact I'm surprised that this lint didn't catch, e.g., [Unicode](https://master-docs-flutter-io.firebaseapp.com/flutter/foundation/Unicode-class.html)...

...oh, I see. It's because it has a private constructor! The PointerEventConverter class should have a private `_` constructor too, to prevent you from creating one.

Maybe the lint should be changed to "classes_with_only_static_members_should_have_private_constructor"...

**Comment by a14n** [comment]
The lint `avoid_classes_with_only_static_members` is to enforce the Effective Dart rule [AVOID defining a class that contains only static members.](https://www.dartlang.org/guides/language/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members).

The private constructor looks like a kind of unexpected workaround to ignore the rule :)

**Comment by Hixie** [comment]
Ah, yeah, we don't really care about that rule. We do care about not having classes that have only static methods _and_ can be constructed, though. So we should keep the lint, it's flagging a real bug.

**Comment by a14n** [comment]
Should I take your comment as a _LGTM_ ?

**Comment by Hixie** [comment]
LGTM

**Pull request merged**